init-tools.sh

```diff
@@ -12,116 +12,109 @@ __BUILD_TOOLS_PACKAGE_VERSION=$(cat $__scriptpath/BuildToolsVersion.txt)
 __DOTNET_TOOLS_VERSION=$(cat $__scriptpath/DotnetCLIVersion.txt)
 __BUILD_TOOLS_PATH=$__PACKAGES_DIR/microsoft.dotnet.buildtools/$__BUILD_TOOLS_PACKAGE_VERSION/lib
 __INIT_TOOLS_RESTORE_PROJECT=$__scriptpath/init-tools.msbuild
-__INIT_TOOLS_DONE_MARKER_DIR=$__TOOLRUNTIME_DIR/$__BUILD_TOOLS_PACKAGE_VERSION
-__INIT_TOOLS_DONE_MARKER=$__INIT_TOOLS_DONE_MARKER_DIR/done
+__BUILD_TOOLS_SEMAPHORE=$__TOOLRUNTIME_DIR/$__BUILD_TOOLS_PACKAGE_VERSION/init-tools.complete
 
-if [ -z "$__DOTNET_PKG" ]; then
-    if [ "$(uname -m | grep "i[3456]86")" = "i686" ]; then
-        echo "Warning: build not supported on 32 bit Unix"
-    fi
-OSName=$(uname -s)
-    case $OSName in
-        Darwin)
-            OS=OSX
-            __DOTNET_PKG=dotnet-dev-osx-x64
-            ulimit -n 2048
-            ;;
-
-        Linux)
-            __DOTNET_PKG=dotnet-dev-linux-x64
-            OS=Linux
-
-            if [ -e /etc/redhat-release ]; then
-                redhatRelease=$(</etc/redhat-release)
-                if [[ $redhatRelease == "CentOS release 6."* || $redhatRelease == "Red Hat Enterprise Linux Server release 6."* ]]; then
-                    __DOTNET_PKG=dotnet-dev-rhel.6-x64
-                fi
-            fi
+if [ -e $__BUILD_TOOLS_SEMAPHORE ]; then
+    echo "Tools are already initialized"
+    return #return instead of exit because this script is inlined in other scripts which we don't want to exit
+fi
 
-            ;;
+if [ -e $__TOOLRUNTIME_DIR ]; then rm -rf -- $__TOOLRUNTIME_DIR; fi
 
-        *)
-            echo "Unsupported OS '$OSName' detected. Downloading linux-x64 tools."
-            OS=Linux
-            __DOTNET_PKG=dotnet-dev-linux-x64
-            ;;
-  esac
-fi
+if [ -d "$DotNetBuildToolsDir" ]; then
+    echo "Using tools from '$DotNetBuildToolsDir'."
+    ln -s "$DotNetBuildToolsDir" "$__TOOLRUNTIME_DIR"
```

Comment:
So the new way of using own build of tools is to set the DotNetBuildToolsDir instead of DOTNET_TOOLSET_DIR, right?

Reply:
Correct.  As part of the "build from source" effort, we are working on standardizing our tools so that when we build all repos together, they share a single place for the tools.